### PR TITLE
fix(react-component): updating TC method to use echarts's methods

### PR DESCRIPTION
**Pull request by mnischay**
## Overview
updating the calculateTimeStamp method to use echarts's for obtaining timestamp of click
1. adding echarts to dashboard's bar and scatter charts
2. updating the calculateTimeStamp method to use echarts's for obtaining timestamp of click
3. using chartRef instead of ref to within trendcursors
4. adding interpolation for different step types

## Verifying Changes
refer to screen capture

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/customization/widgets/barChart/component.tsx [hunk 1]
```diff
@@ -33,6 +34,27 @@ const BarChartWidgetComponent: React.FC<BarChartWidget> = (widget) => {
   const aggregation = getAggregation(queryConfig);
 
   const significantDigits = widgetSignificantDigits ?? dashboardSignificantDigits;
+  const grid = useSelector((state: DashboardState) => state.grid);
+  const size = { width: grid.cellSize * widget.width, height: grid.cellSize * widget.height };
+
+  if (showEcharts) {
+    return (
+      <Chart
```
Comment (mnischay) [inline packages/dashboard/src/customization/widgets/barChart/component.tsx:42]: we should probably have a generic widget component and this component might not be even required...

File: packages/react-components/src/components/chart/utils/getInfo.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { DurationViewport, Viewport } from '@iot-app-kit/core/src';
```
Comment (corteggiano) [inline packages/react-components/src/components/chart/utils/getInfo.ts:1]: is this whole file doing interpolation calculations? if so could we rename it to something less generic... getInfo doesn't tell us much about what it does

Comment (mnischay) [reply packages/react-components/src/components/chart/utils/getInfo.ts:1]: Sure thing, any suggestions ? , it is all the auxiliary functions we have for charts / TCs...

File: packages/dashboard/src/customization/widgets/barChart/component.tsx (hunk 1 above)
Comment (corteggiano) [reply packages/dashboard/src/customization/widgets/barChart/component.tsx:42]: we could split each out into a separate file in the utils folder, or move the TC/interpolation helpers closer to where they're used

File: packages/react-components/src/components/chart/utils/getInfo.ts (hunk 2 above)
Comment (corteggiano) [reply packages/react-components/src/components/chart/utils/getInfo.ts:1]: we could split each out into a separate file in the utils folder, or move the TC/interpolation helpers closer to where they're used

**Review by corteggiano — APPROVED** [review]
looks good! lets go TCs 🥳

**Pull request merged**